orchestra/analytics/latency.py

```diff
@@ -0,0 +1,112 @@
+        iteration: the work iteration for that assignment.
+            - assignment 0 iteration 0 represents the period from task creation
+              to first assignment.
+            - assignment N > 0 iteration 0 represents the period from previous
+              submission to reviewer assignment.
+        start_datetime: the time the iteration started.
+        calendar_time: the wallclock time it took for the worker to complete
+            that iteration.
+        work_time: the amount of work time the worker reported dedicating to
+            that iteration (0 for iteration 0).
+
+    Args:
+        projects (django.db.models.query.QuerySet):
+            An iterable of projects to include in the report.
+
+    Returns:
+        df (pandas.DataFrame):
+            A DataFrame with timing information.
+    """
+    # TODO(marcua): `get_project_information` per project means we do random
+    # access all over the database.  Ideally, we'd have a sequential scan
+    # version of this.
+    time_data = list(itertools.chain.from_iterable(
+        project_time_row_generator(
+            get_project_information(project.id))
+        for project in projects))
+    df = DataFrame(time_data)
+    df = df[
+        ['project', 'task', 'assignment', 'iteration', 'start_datetime',
+         'calendar_time', 'work_time']]
+    return df
+
+
+class Iteration(object):
+    """ Represents a task assignment iteration's data for reporting. """
+    def __init__(self, project, task, assignment, iteration, work_time,
+                 end_datetime):
+        self.project = project
+        self.task = task
+        self.assignment = assignment
+        self.iteration = iteration
+        self.work_time = work_time
+        self.end_datetime = end_datetime
+
+    def to_dict(self, start_datetime):
+        return {
+            'project': self.project['id'],
+            'task': self.task['id'],
+            'assignment': self.assignment,
+            'iteration': self.iteration,
+            'start_datetime': start_datetime,
+            'work_time': self.work_time,
+            'calendar_time': self.end_datetime - start_datetime
+        }
+
+
+def project_time_row_generator(project_information):
+    """
+    Given a project, yields all task assignment iterations, reporting
+    calendar and work times.
+    """
+    # lol at this variable name
+    human_slugs = {
+        step['slug'] for step in project_information['steps']
+        if step['worker_type'] == Step.WorkerType.HUMAN}
+
+    for step_slug, task in iter(project_information['tasks'].items()):
+        if step_slug not in human_slugs:
+            continue
+        if task['status'] != dict(Task.STATUS_CHOICES)[Task.Status.COMPLETE]:
+            continue
+        iterations = []
+        for assignment_idx, assignment in enumerate(task['assignments']):
+            for iteration, snapshot in enumerate(
+                    assignment['snapshots']['snapshots']):
```

Comment:
yea so snapshots is a field that requires a jsonfield, which is itself a dict :/